apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/contacts/ContactsView.kt

```diff
@@ -0,0 +1,433 @@
+import chat.simplex.common.platform.getKeyboardState
+import chat.simplex.common.platform.hideKeyboard
+import chat.simplex.common.ui.theme.DEFAULT_PADDING
+import chat.simplex.common.ui.theme.DEFAULT_PADDING_HALF
+import chat.simplex.common.views.chatlist.connect
+import chat.simplex.common.views.helpers.AppBarTitle
+import chat.simplex.common.views.helpers.KeyChangeEffect
+import chat.simplex.common.views.helpers.KeyboardState
+import chat.simplex.common.views.helpers.ModalData
+import chat.simplex.common.views.helpers.ModalManager
+import chat.simplex.common.views.helpers.ModalView
+import chat.simplex.common.views.helpers.SearchTextField
+import chat.simplex.common.views.helpers.generalGetString
+import chat.simplex.common.views.helpers.hostDevice
+import chat.simplex.common.views.newchat.strHasSingleSimplexLink
+import chat.simplex.res.MR
+import dev.icerock.moko.resources.compose.painterResource
+import dev.icerock.moko.resources.compose.stringResource
+import kotlinx.coroutines.flow.distinctUntilChanged
+
+enum class ContactType {
+    CARD, REQUEST, RECENT, REMOVED, UNKNOWN
+}
+
+private fun contactChats(c: List<Chat>, contactTypes: List<ContactType>): List<Chat> {
+    return c.filter { chat -> contactTypes.contains(getContactType(chat)) }
+}
+
+fun getContactType(chat: Chat): ContactType {
+    return when (val cInfo = chat.chatInfo) {
+        is ChatInfo.ContactRequest -> ContactType.REQUEST
+        is ChatInfo.Direct -> {
+            val contact = cInfo.contact;
+
+            when {
+                contact.activeConn == null && contact.profile.contactLink != null -> ContactType.CARD
+                contact.chatDeleted -> ContactType.REMOVED
+                contact.contactStatus != ContactStatus.DeletedByUser && contact.contactStatus != ContactStatus.Deleted -> ContactType.RECENT
+                else -> ContactType.UNKNOWN
+            }
+        }
+        else -> ContactType.UNKNOWN
+    }
+}
+
+private val chatsByTypeComparator = Comparator<Chat> { chat1, chat2 ->
+    val chat1Type = getContactType(chat1)
+    val chat2Type = getContactType(chat2)
+
+    when {
+        chat1Type.ordinal < chat2Type.ordinal -> -1
+        chat1Type.ordinal > chat2Type.ordinal -> 1
+
+        else -> chat2.chatInfo.chatTs.compareTo(chat1.chatInfo.chatTs)
+    }
+}
+
+@Composable
+private fun ModalData.DeletedContactsView(rh: RemoteHostInfo?, close: () -> Unit) {
+    ModalView(
+        close = close
+    ) {
+        Column(
+            Modifier.fillMaxSize(),
+        ) {
+            Box(contentAlignment = Alignment.Center) {
+                val bottomPadding = DEFAULT_PADDING
+                AppBarTitle(
+                    stringResource(MR.strings.chat_deleted),
+                    hostDevice(rh?.remoteHostId),
+                    bottomPadding = bottomPadding
+                )
+            }
+
+            ContactsLayout(
+                contactActions = {},
+                contactTypes = listOf(ContactType.REMOVED),
```

Comment:
opening deleted conversations (and even writing message) doesn't keep contact in chat list - we can do a simple fix by doing this for contacts with deleted conversation when opening chat with them:

```
        if (contact.chatDeleted) {
          chatModel.updateContact(chat.remoteHostId, contact.copy(chatDeleted = false))
        }
```

(variable names may differ - took it from a different place)

This way it would stay in chat list for current session even if user doesn't write message immediately - they can still re-open from chat list and write later. After app restart it won't be in list, but I think it's ok edge case - can be consider as "user didn't have intention to restore contact" (otherwise we'd have to do an api call to mark chatDeleted true in backend on open, which I think is unnecessary unless somebody complains)